x-pack/legacy/plugins/apm/server/lib/transactions/breakdown/index.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { flatten } from 'lodash';
+import { PromiseReturnType } from '../../../../typings/common';
+import { Setup } from '../../helpers/setup_request';
+import { rangeFilter } from '../../helpers/range_filter';
+
+export type TransactionBreakdownAPIResponse = PromiseReturnType<
+  typeof getTransactionBreakdown
+>;
+
+export async function getTransactionBreakdown({
+  setup,
+  serviceName,
+  transactionName
+}: {
+  setup: Setup;
+  serviceName: string;
+  transactionName?: string;
+}) {
+  const { uiFiltersES, client, config, start, end } = setup;
+
+  const params = {
+    index: config.get<string>('apm_oss.metricsIndices'),
+    body: {
+      size: 0,
+      query: {
+        bool: {
+          must: [
+            {
+              term: {
+                'service.name': {
+                  value: serviceName
+                }
+              }
+            },
+            {
+              term: {
+                'transaction.type': {
+                  value: 'request'
+                }
+              }
+            },
+            { range: rangeFilter(start, end) },
+            ...uiFiltersES,
+            ...(transactionName
+              ? [
+                  {
+                    term: {
+                      'transaction.name': {
+                        value: transactionName
+                      }
+                    }
+                  }
+                ]
+              : [])
+          ]
+        }
+      },
+      aggs: {
+        sum_all_self_times: {
+          sum: {
+            field: 'span.self_time.sum'
+          }
+        },
+        total_transaction_breakdown_count: {
+          sum: {
+            field: 'transaction.breakdown.count'
+          }
+        },
+        types: {
+          terms: {
+            field: 'span.type',
+            size: 42
```

Comment:
what's the significance of `size: 42`?

Reply:
It's from @felixbarny's example query. I'm honestly not sure what amount we can expect here. I'll ask.

Reply:
I've changed this to 20x20, and we cut off at 20 items, per a discussion on Slack in the breakdown graphs channel.